openlibrary/templates/books/breadcrumb_select.html

```diff
@@ -0,0 +1,68 @@
+$def with (key=None, selected="", options=None)
+    $if not options:
+        $selected
+    $elif len(options) == 1:
+        $if selected != options[0][0]:
+            <span class="crumb">
+                <span class="crumbicon">
+                    <div class="dropdown">
+                        <h2 class="breadcrumb-title">$selected</h2>
+                        <img class="icon-link__image down-chevron" src="/static/images/icons/right-chevron.svg">
+                    </div>
+                    <select class="disguised-select">
+                        <option selected>$selected</option>
+                        <option value="$options[0][1]">$options[0][0]</option>
+                    </select>
+                </span>
+            </span>
+        $else:
+            <h1>$selected</h1>
+    $else:
+        $ breadcrumb_title = ""
+        $for (title, url) in options:
+            $if '(' in title and title[:title.index('(')-1] == selected:
+                $ breadcrumb_title = title
+            $elif title == selected:
+                $ breadcrumb_title = title;
+        <span class="crumb">
+            <span class="crumbicon">
+                <div class="dropdown">
+                    <h2 class="breadcrumb-title">$breadcrumb_title</h2>
+                    <img class="icon-link__image down-chevron" src="/static/images/icons/right-chevron.svg">
+                </div>
+                <select class="disguised-select">
+                    $for (title, url) in options:
+                        $if '(' in title and title[:title.index('(')-1] == selected:
+                            <option selected>$title</option>
+                        $elif title == selected:
+                            <option selected>$title</option>
+                        $else:
+                            <option value=$url>$title</option>
+                </select>
+            </span>
+        </span>
+
+        <script>
+            const crumbs = document.querySelectorAll('.crumb select')
+            const allowedKeys = new Set(['Tab', 'Enter', ' '])
+            const preventedKeys = new Set(['ArrowUp', 'ArrowDown'])
+
+            // watch crumbs for changes,
+            // ensures it's a full value change, not a user exploring options via keyboard
+            crumbs.forEach(nav => {
+                let ignoreChange = false
+
+                nav.addEventListener('change', e => {
+                    if (ignoreChange) return
+                    // it's actually changed!
+                    window.location = nav.value;
+                })
+
+            nav.addEventListener('keydown', ({ key }) => {
+                if (preventedKeys.has(key))
+                ignoreChange = true
+                else if (allowedKeys.has(key))
+                ignoreChange = false
+            })
+            })
```

Comment:
@szgrune, can you explain the desired behavior here?  The selected breadcrumb wraps in FF, but does not in Chrome.  Also, what it the `' '` key, and how do I test it?